debian/lxd.preinst

```
@@ -23,12 +23,12 @@ case "$1" in
         COUNT=0
         SKIP=false
         while :; do
-            if [ ! -e "/run/snapd.socket" ] && [ ! -e "/var/lib/lxd/server.crt" ] \ 
-                || [ ! -e "/run/snapd.socket" ] \
-                && [ ! "$(ls -A /var/lib/lxd/containers)" ] \
-                && [ ! "$(ls -A /var/lib/lxd/images)" ] \
-                && [ ! "$(ls -A /var/lib/lxd/networks)" ] \
-                && [ ! "$(ls -A /var/lib/lxd/storage-pools)" ]
+            if ( [ ! -e "/run/snapd.socket" ] && [ ! -e "/var/lib/lxd/server.crt" ] ) \
+                || ( [ ! -e "/run/snapd.socket" ] \
+                && [ -z "$(ls -A /var/lib/lxd/containers)" ] \
+                && [ -z "$(ls -A /var/lib/lxd/images)" ] \
+                && [ -z "$(ls -A /var/lib/lxd/networks)" ] \
+                && [ -z "$(ls -A /var/lib/lxd/storage-pools)" ] )
```

What are you doing for the case where those don't exist?

---

Given the complexity here, it feels like having a function doing the check is probably going to be cleaner.

---

Right now I have used `>/dev/null 2>&1` in the commit https://github.com/lxc/lxd-pkg-ubuntu/pull/39/commits/28e93db865d25a1cd5a1c4c7cc4cd5328a7c68ef, would this be a acceptable way to do that?

---

No, now you're suppressing all output while not ignoring errors, so you just made it significantly worse by always having those tests succeed.

---

Thank you. Right now I have reimplemented the logic completely